modyn/tests/selector/internal/trigger_sample/test_trigger_sample_storage.py

```diff
@@ -154,42 +154,35 @@ def test_get_trigger_samples():
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 0, 4, 16)
     assert len(result) == 4
-    assert result == [expected_order[0], expected_order[1], expected_order[2], expected_order[3]]
+    assert result == expected_order[0:4]
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 1, 4, 16)
     assert len(result) == 4
-    assert result == [expected_order[4], expected_order[5], expected_order[6], expected_order[7]]
+    assert result == expected_order[4:8]
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 2, 4, 16)
     assert len(result) == 4
-    assert result == [expected_order[8], expected_order[9], expected_order[10], expected_order[11]]
+    assert result == expected_order[8:12]
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 3, 4, 16)
     assert len(result) == 4
-    assert result == [expected_order[12], expected_order[13], expected_order[14], expected_order[15]]
+    assert result == expected_order[12:16]
 
-    result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 1, 3, 16)
+    result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 0, 3, 16)
     assert len(result) == 6
-    assert result == [
-        expected_order[6],
-        expected_order[7],
-        expected_order[8],
-        expected_order[9],
-        expected_order[10],
-        expected_order[11],
-    ]
+    assert result == expected_order[0:6]
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 2, 3, 16)
-    assert len(result) == 4
-    assert result == [expected_order[12], expected_order[13], expected_order[14], expected_order[15]]
+    assert len(result) == 5
+    assert result == expected_order[11:16]
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 0, 1, 16)
     assert len(result) == 16
     assert result == expected_order
 
     result = TriggerSampleStorage(TMP_DIR).get_trigger_samples(1, 2, 3, 9, 10, 16)
-    assert len(result) == 0
-    assert not result
+    assert len(result) == 1
+    assert result == [expected_order[15]]
```

Comment:
To give some context, we are checking the result of the last worker [9] if we have a total of 16 elements and 10 workers. The existing version gives 0 elements to the last worker (and 2 to workers [0, ..., 7], 0 to worker [8]). My version assigns 2 elements to workers [0, ..., 5] and 1 element to workers [6, 7, 8, 9].

Reply:
Alright, thanks for explaining that again. I am not actually sure what is the better strategy here but let's just keep yours, since it is more balanced overall (could also be valid to say that we use more workers to their maximum in the other version, but again not sure)